packages/passport-interface/src/PassportPopup.ts

```diff
@@ -0,0 +1,80 @@
+import { useEffect, useState } from "react";
+
+/**
+ * React hook that listens for PCDs and PendingPCDs from a passport popup window
+ * using message passing and event listeners.
+ */
+export function usePassportPopupMessages() {
+  const [pcdStr, setPCDStr] = useState("");
+  const [pendingPCDStr, setPendingPCDStr] = useState("");
+
+  // Listen for PCDs coming back from the Passport popup
+  useEffect(() => {
+    function receiveMessage(ev: MessageEvent<any>) {
+      // Extensions including Metamask apparently send messages to every page. Ignore those.
+      if (ev.data.encodedPCD) {
+        console.log("Received PCD", ev.data.encodedPCD);
+        setPCDStr(ev.data.encodedPCD);
+      } else if (ev.data.encodedPendingPCD) {
+        console.log(ev.data);
+        setPendingPCDStr(ev.data.encodedPendingPCD);
+      }
+    }
+    window.addEventListener("message", receiveMessage, false);
+    return () => window.removeEventListener("message", receiveMessage);
+  }, []);
+
+  return [pcdStr, pendingPCDStr];
+}
+
+/**
+ * React hook to be used on whatever route the passport popup is redirected from.
+ * Sends messages back to original page to be processed by `usePassportPopupMessages`.
+ */
+export function usePassportPopupSetup() {
```

Comment:
Something about the name rubs me the wrong way - I think `Setup` doesn't make a lot of sense to me. Maybe something like `userWaitForPopup()` could work better?